lib/Dialect/FIRRTL/Transforms/Dedup.cpp

```diff
@@ -1705,9 +1716,6 @@ class DedupPass : public circt::firrtl::impl::DedupBase<DedupPass> {
               ext && !ext.getDefname().has_value())
             return success();
 
-          if (!checkVisibility(module))
-            return success();
```

Comment:
Something to keep an eye on: now that there is no early exit here, all modules will be hashed. This may affect performance, specifically for private classes.